engine/classes/Elgg/Http/Request.php

```diff
@@ -417,7 +417,7 @@ public function validate() {
 		// Requests with multipart content type
 		$post_data_count = count($this->request->all());
 		// Requests with other content types
-		$post_body_length = is_string($this->content) ? strlen($this->content) : 0;
+		$post_body_length = is_string($this->getContent()) ? strlen($this->content) : 0;
```

Comment:
shouldn't the `strlen` also be based on `->getContent()`

Reply:
yeah

Reply:
Thought it doesn't matter. $this->content is populated after $this->getContent() is called. If there is no raw body, it reads the buffer and sets the content

Reply:
if $this->content is a resource to begin with you need to use getContent() for the strlen otherwise i don't know what error could occur.

Reply:
never mind i see you changed it